cores/arduino/USBSerial.cpp

```diff
@@ -83,9 +83,11 @@ size_t USBSerial::write(const uint8_t *buffer, size_t size)
       buffer += portion;
       // After storing data, start transmitting process
       CDC_continue_transmit();
+    } else if (!CDC_connected()) {
```

Comment:
Doing further investigation.
I think it should be better to put the "CDC_connected() " in the while condition
` while (CDC_connected() && (rest > 0)) {`

Reply:
@ghent360 did you see this comment?

Reply:
When you move it to the while. When I disconnect and re-connect the USB the USB serial sometimes stops transmitting. I believe it goes in perpetual timeout state, because we stop sending data and the timeout is never reset.

Reply:
I added core to reset the transmit timeout when the port is re-oppened.

Reply:
Make sense, thanks.